google/cloud/internal/grpc_opentelemetry_test.cc

```diff
@@ -208,6 +210,52 @@ TEST(OpenTelemetry, TracedAsyncBackoffDisabled) {
   EXPECT_THAT(spans, IsEmpty());
 }
 
+TEST(OpenTelemetry, TracedAsyncBackoffPreservesContext) {
+  if (CompilerId() == "GNU" && CompilerVersion() == "7.3.1") {
```

Comment:
Does OTel claim to support GCC 7.x?  Is there a bug in OTel we can reference?  Can we detect this with an OTel version?

Reply:
> Does OTel claim to support GCC 7.x?

My reading is they support compilers that implement the standard correctly.

> Is there a bug in OTel we can reference?

good idea, and yes: https://github.com/open-telemetry/opentelemetry-cpp/issues/1014

Reply:
There is a patch linked in the issue. I will try to apply it to our `centos-7` dockerfile.

Reply:
According to some comments on that bug, GCC 8.x also trouble. You may need to apply that patch to several other builds.

Reply:
> My reading is they support compilers that implement the standard correctly.

There are no compilers without defects.  That implies they support no compilers at all.

Reply:
Patched.

> According to some comments on that bug, GCC 8.x also trouble. You may need to apply that patch to several other builds.

Thanks for pointing that out. I forgot that the `demo-install` builds do not run these unit tests. I assumed they all worked.

Opened #13159 because I am not sure if we should apply the patch by default, or just call out the issue in a comment.

> There are no compilers without defects. That implies they support no compilers at all.

I realize that and my comment was tongue-in-cheek. Here is their actual text: https://github.com/open-telemetry/opentelemetry-cpp/blob/cb603ad97f33e52340e627e1cb43ba73bb1d7ef0/README.md?plain=1#L48-L49